AntWars/GamePanel.cs

```diff
@@ -57,95 +57,140 @@ private void print(IList<BoardObject> boardObjects)
             }
+
             bitmap = new Bitmap(bitmap, bitmap.Width * 4, bitmap.Height * 4);
             pb_Game.Image = bitmap;
         }
 
+        private void setColor(BoardObject obj, Color inferiorElement, Bitmap bitmap)
+        {
+            if (obj.isCarry())
+            {
+                setColorCarry(obj, inferiorElement, bitmap);
+            }
+            else if (obj.isScout())
+            {
+                setColorScout(obj, inferiorElement, bitmap);
+            }
+            else if (obj.isSugar())
+            {
+                setColorSugar(obj, inferiorElement, bitmap);
+            }
+            else if (obj.isBase())
+            {
+                setColorBase(obj, inferiorElement, bitmap);
+            }
+            else if (obj.isSignal())
+            {
+                setColorSignal(obj, inferiorElement, bitmap);
+            }
+        }
+
+        private void setColorCarry(BoardObject obj, Color inferiorElement, Bitmap bitmap)
+        {
+            if (((Ant)obj).Owner == game.Player1)
+            {
+                if (!string.IsNullOrEmpty(inferiorElement.Name) && inferiorElement == Color.Black) //wenn Sugar drunter ist
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.DarkGreen);
+                else
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.Green);
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(inferiorElement.Name) && inferiorElement == Color.Black) //wenn Sugar drunter ist
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.DarkBlue);
+                else
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.Blue);
+            }
+        }
+
+        private void setColorScout(BoardObject obj, Color inferiorElement, Bitmap bitmap)
+        {
+            if (((Scout)obj).Owner == game.Player1)
+            {
+                if (!string.IsNullOrEmpty(inferiorElement.Name) && inferiorElement == Color.Black) //wenn Sugar drunter ist
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.DarkSeaGreen);
+                else
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.SeaGreen);
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(inferiorElement.Name) && inferiorElement == Color.Black) //wenn Sugar drunter ist
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.DarkSlateBlue);
+                else
+                    bitmap.SetPixel(obj.Coords.X, obj.Coords.Y, System.Drawing.Color.SlateBlue);
+            }
+        }
+
+        private void setColorSugar(BoardObject obj, Color inferiorElement, Bitmap bitmap)
+        {
+            switch (inferiorElement.Name)
+            {
+                case ("ff008000"): //Player1-Carry
```

Comment:
Wäre im code anschaulicher wenn das statische constanten sind, ist nur ein nice to have